src/wallet/wallet.cpp

```diff
@@ -1448,7 +1448,7 @@ void CWallet::BlockUntilSyncedToCurrentChain() {
         // We could also take cs_wallet here, and call m_last_block_processed
         // protected by cs_wallet instead of cs_main, but as long as we need
         // cs_main here anyway, it's easier to just call it cs_main-protected.
-        LOCK(cs_main);
+        LOCK2(cs_main, cs_wallet);
```

Comment:
With the above comment, is this necessary?

Reply:
Good catch; yea.  It would take some digging to see if cs_wallet is actually needed, or to do the action called out for.